int64 particles
PBPF/object_pose[] objects
